mmdet/engine/hooks/visualization_hook.py

```diff
@@ -153,3 +154,161 @@ def after_test_iter(self, runner: Runner, batch_idx: int, data_batch: dict,
+    3. ``vis_backends`` takes effect if the user does not specify ``show``
+        and `test_out_dir``. You can set ``vis_backends`` to WandbVisBackend or
+        TensorboardVisBackend to store the prediction result in Wandb or
+        Tensorboard.
+
+    Args:
+        draw (bool): whether to draw prediction results. If it is False,
+            it means that no drawing will be done. Defaults to False.
+        interval (int): The interval of visualization. Defaults to 30.
+        score_thr (float): The threshold to visualize the bboxes
+            and masks. Defaults to 0.3.
+        show (bool): Whether to display the drawn image. Default to False.
+        wait_time (float): The interval of show (s). Defaults to 0.
+        test_out_dir (str, optional): directory where painted images
+            will be saved in testing process.
+        backend_args (dict): Arguments to instantiate a file client.
+            Defaults to ``None``.
+    """
+
+    def __init__(self,
+                 draw: bool = False,
+                 interval: int = 1,
+                 score_thr: float = 0.3,
+                 show: bool = False,
+                 wait_time: float = 0.,
+                 test_out_dir: Optional[str] = None,
+                 backend_args: dict = None):
+        self._visualizer: Visualizer = Visualizer.get_current_instance()
+        self.interval = interval
+        self.score_thr = score_thr
+        self.show = show
+        if self.show:
+            # No need to think about vis backends.
+            self._visualizer._vis_backends = {}
+            warnings.warn('The show is True, it means that only '
+                          'the prediction results are visualized '
+                          'without storing data, so vis_backends '
+                          'needs to be excluded.')
+
+        self.wait_time = wait_time
+        self.backend_args = backend_args
+        self.draw = draw
+        self.test_out_dir = test_out_dir
+        self.image_idx = 0
+
+    def after_val_iter(self, runner: Runner, batch_idx: int, data_batch: dict,
+                       outputs: Sequence[TrackDataSample]) -> None:
+        """Run after every ``self.interval`` validation iteration.
+
+        Args:
+            runner (:obj:`Runner`): The runner of the validation process.
+            batch_idx (int): The index of the current batch in the val loop.
+            data_batch (dict): Data from dataloader.
+            outputs (Sequence[:obj:`TrackDataSample`]): Outputs from model.
+        """
+        if self.draw is False:
+            return
+
+        assert len(outputs) == 1,\
+            'only batch_size=1 is supported while validating.'
+
+        sampler = runner.val_dataloader.sampler
+        if isinstance(sampler, ImgQuotaSampler):
+            if self.every_n_inner_iters(batch_idx, self.interval):
+                track_data_sample = outputs[0]
+                self.visualize_single_image(track_data_sample[0], batch_idx)
```

Comment:
The `step` in  `self.visualize_single_image` should be global throughout the training process. https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/engine/hooks/visualization_hook.py#L96